src/gui/XournalppCursor.cpp

```diff
@@ -118,119 +132,127 @@ void XournalppCursor::updateCursor()
 	XOJ_CHECK_TYPE(XournalppCursor);
 
 	MainWindow* win = control->getWindow();
-	if (!win)
-	{
-		return;
-	}
+	if (!win) return;
 
 	XournalView* xournal = win->getXournal();
-	if (!xournal)
-	{
-		return;
-	}
+	if (!xournal) return;
 
 	GdkWindow* window = gtk_widget_get_window(win->getWindow());
 
 	GdkCursor* cursor = NULL;
-
+	bool useDefault = false;
+	
+	
 	if (this->busy)
 	{
-		cursor = gdk_cursor_new_for_display(gdk_window_get_display(window), GDK_WATCH);
+		setStockCursor(GDK_WATCH);
```

Comment:
The combination of `setStockCursor` and `cursor = ...` is error prone. I suggest refactoring this so that only one of `setStockCursor` or `cursor = ...` is used.